src/fallback.rs

```diff
@@ -100,7 +101,7 @@ pub fn fish_wcswidth(s: &wstr) -> isize {
 // Replacement for mkostemp(str, O_CLOEXEC)
 // This uses mkostemp if available,
 // otherwise it uses mkstemp followed by fcntl
-pub fn fish_mkstemp_cloexec(name_template: CString) -> (RawFd, CString) {
+pub fn fish_mkstemp_cloexec(name_template: CString) -> nix::Result<(OwnedFd, CString)> {
```

Comment:
The reason why I used `Result<(Fd, Path)>` signature, rather than putting the path outside of the result (like the existing usage of this function would suggest) is described in https://github.com/fish-shell/fish-shell/pull/10308/files#r1493507686